### PR TITLE
Implement dot operator

### PR DESCRIPTION
**Pull request by lysnikolaou**
Implement dot operator, which gathers the parsed nodes into a sequence, ignoring their separators(such as a comma):
    * Add node in grammar.py
    * Generate correct Python code for new node

I didn't get to implement the generation of C code for this, but I'd like to have a 'go' or 'no go' before I do, regarding the way I implemented this. Could it be maybe done better?

### CODE REVIEW

File: pegen/python_generator.py [hunk 1]
```diff
@@ -107,6 +108,13 @@ def visit_Repeat1(self, node: Repeat1) -> Tuple[str, str]:
         self.cache[node] = name, f"self.{name}()"  # But no trailing comma here!
         return self.cache[node]
 
+    def visit_RepeatWithSeparator(self, node: RepeatWithSeparator) -> Tuple[str, str]:
+        if node in self.cache:
+            return self.cache[node]
+        name = self.gen.name_node_with_sep(node)
+        self.cache[node] = name, f"self.{name}()"  # But no trailing comma here!
```
Comment (gvanrossum) [inline pegen/python_generator.py]: I'd change this comment to "# No trailing comma here either!" (so the sequence of comments in these 4 methods reads somewhat humorously).

File: pegen/grammar.py [hunk 2]
```diff
@@ -72,6 +72,9 @@ def __init__(self, name: str, type: Optional[str], rhs: Rhs):
     def is_loop(self) -> bool:
         return self.name.startswith("_loop")
 
+    def is_node_with_sep(self) -> bool:
```
Comment (gvanrossum) [inline pegen/grammar.py]: I'm not excited about the long name here (esp. since it's repeated in a keyword arg later).

**Comment by gvanrossum** [comment]
I guess now we can start using it...

**Comment by lysnikolaou** [comment]
I am currently working on implementing the C code generation for this. I thought that this would go into this PR. Should I create a new one?

**Comment by gvanrossum** [comment]
Oops. I guess we need a convention: if you put [WIP] in the subject of a PR
that means not to merge it yet even if it's perfect. Go ahead and make a
new PR here (I like smaller steps anyway, but I like squashing PRs since
the local commits are rarely of historical value later).

**Comment by lysnikolaou** [comment]
I'll do that in the future then.

**Pull request merged**